service/src/main/kotlin/fi/espoo/evaka/invoicing/service/FeeDecisionService.kt

```diff
@@ -61,21 +62,27 @@ class FeeDecisionService(
         tx: Database.Transaction,
         user: AuthenticatedUser,
         ids: List<FeeDecisionId>,
-        now: Instant
+        lastPossibleDecisionValidFrom: Instant
     ): List<FeeDecisionId> {
         tx.lockFeeDecisions(ids)
         val decisions = tx.getFeeDecisionsByIds(ids)
         if (decisions.isEmpty()) return listOf()
-
+        val now = Instant.now()
         val notDrafts = decisions.filterNot { it.status == DRAFT }
         if (notDrafts.isNotEmpty()) {
             throw BadRequest("Some fee decisions were not drafts")
         }
 
-        val nowDate = LocalDate.from(now.atZone(helsinkiZone))
-        val notYetValidDecisions = decisions.filter { it.validFrom > nowDate }
-        if (notYetValidDecisions.isNotEmpty()) {
-            throw BadRequest("Some of the fee decisions are not valid yet")
+        val lastPossibleDecisionValidFromDate = LocalDate.from(lastPossibleDecisionValidFrom
+            .atZone(helsinkiZone))
+        val decisionsNotValidForConfirmation = decisions.filter {
+            it.validFrom.atStartOfDay() > lastPossibleDecisionValidFromDate.atStartOfDay()
```

Comment:
It doesn't seem like converting `LocalDate`s to `LocalDateTime`s has any benefit here.